File: 2ª Unidade/Normalização/frequencia.pl

% Relação Frequencia

frequencia([idTurma, idPessoa, data, estavaFrequente]).

fds([ [[idTurma, idPessoa, data], [estavaFrequente]] ]).


% Query :   frequencia(R), fds(F), threenf(R, F, T)


% Resultado: R == T

% F = [[[idTurma, idPessoa, data], [estavaFrequente]]],
% R = [idTurma, idPessoa, data, estavaFrequente],
% T = [[idTurma, idPessoa, data, estavaFrequente]]

File: Desenvolvimento/Back/src/entidades/Frequencia.java
package entidades;

import java.sql.Date;

public class Frequencia {
	private int idPessoa;
	private int idTurma;
	private Date data;
	private boolean estavaFrequente;
	
	public Frequencia(int idPessoa, int idTurma, Date data, boolean estavaFrequente) {
		super();
		this.idPessoa = idPessoa;
		this.idTurma = idTurma;
		this.data = data;
		this.estavaFrequente = estavaFrequente;
	}

	public Frequencia() {
		super();
	}

	public int getIdPessoa() {
		return idPessoa;
	}

	public void setIdPessoa(int idPessoa) {
		this.idPessoa = idPessoa;
	}

	public int getIdTurma() {
		return idTurma;
	}

	public void setIdTurma(int idTurma) {
		this.idTurma = idTurma;
	}

	public Date getData() {
		return data;
	}

	public void setData(Date data) {
		this.data = data;
	}

	public boolean isEstavaFrequente() {
		return estavaFrequente;
	}

	public void setEstavaFrequente(boolean estavaFrequente) {
		this.estavaFrequente = estavaFrequente;
	}
}
